core/src/main/java/com/yahoo/ycsb/generator/AcknowledgedCounterGenerator.java

```diff
@@ -28,16 +28,16 @@
 	static final int WINDOW_SIZE = Integer.rotateLeft(1, 20);
 
 	/** The mask to use to turn an id into a slot in {@link #window}. */
-	private static final int WINDOW_MASK = WINDOW_SIZE - 1;
+	private static final long WINDOW_MASK = WINDOW_SIZE - 1;
 
 	private final ReentrantLock lock;
 	private final boolean[] window;
-	private volatile int limit;
+	private volatile long limit;
 
 	/**
 	 * Create a counter that starts at countstart.
 	 */
-	public AcknowledgedCounterGenerator(int countstart)
```

Comment:
for the generators, could we maintain binary compatibility by leaving in int versions of the methods and marking them Deprecated?